src/node/internal/internal_tls_wrap.ts

```diff
@@ -0,0 +1,714 @@
+): ReturnType<TLSSocketType['getPeerCertificate']> {
+  // Returns an object representing a peer certificate.
+  // This function is not supported.
+  throw new Error('getPeerCertificate is not implemented');
+};
+
+TLSSocket.prototype.getCertificate = function (
+  this: TLSSocket
+): ReturnType<TLSSocketType['getCertificate']> {
+  // Returns an object representing the local certificate.
+  // This function is not supported.
+  return null;
+};
+
+TLSSocket.prototype.getPeerX509Certificate = function (
+  this: TLSSocket,
+  _detailed?: boolean
+): ReturnType<TLSSocketType['getPeerX509Certificate']> {
+  // Returns the peer certificate as an X509 certificate.
+  // This function is not supported.
+  return undefined;
+};
+
+TLSSocket.prototype.getX509Certificate = function (
+  this: TLSSocket
+): ReturnType<TLSSocketType['getX509Certificate']> {
+  // Returns the local certificate as an X509 certificate.
+  // This function is not supported.
+  return undefined;
+};
+
+TLSSocket.prototype.setKeyCert = function (
+  this: TLSSocket,
+  _context: unknown
+): void {
+  // Changing private key and certificate to be used with TCP connection
+  // is not supported due to the limitations of connect() api.
+};
+
+[
+  'getCipher',
+  'getSharedSigalgs',
+  'getEphemeralKeyInfo',
+  'getFinished',
+  'getPeerFinished',
+  'getProtocol',
+  'getSession',
+  'getTLSTicket',
+  'isSessionReused',
+  'enableTrace',
+].forEach((method) => {
+  // @ts-expect-error TS7053 Omitting...
+  TLSSocket.prototype[method] = function (): null {
+    // None of these functions are supported by connect() api.
+    return null;
+  };
+});
+
+type NormalizedConnectionOptions = ConnectionOptions &
+  SocketOptions &
+  TlsOptions & {
+    host?: string;
+    port: number;
+  };
+function normalizeConnectArgs(
+  listArgs: unknown[]
+): [NormalizedConnectionOptions] | [NormalizedConnectionOptions, VoidFunction] {
+  const args = _normalizeArgs(listArgs);
+  const options = args[0] as NormalizedConnectionOptions;
+  const cb = args[1] as VoidFunction | undefined;
+
+  // If args[0] was options, then normalize dealt with it.
+  // If args[0] is port, or args[0], args[1] is host, port, we need to
+  // find the options and merge them in, normalize's options has only
+  // the host/port/path args that it knows about, not the tls options.
+  // This means that options.host overrides a host arg.
+  if (listArgs[1] !== null && typeof listArgs[1] === 'object') {
+    Object.assign(options, listArgs[1]);
+  } else if (listArgs[2] !== null && typeof listArgs[2] === 'object') {
+    Object.assign(options, listArgs[2]);
+  }
+
+  return cb ? [options, cb] : [options];
+}
+
+function onConnectSecure(this: TLSSocket): void {
+  this.authorized = true;
+  this.secureConnecting = false;
+  this.emit('secureConnect');
+
+  this[kIsVerified] = true;
+  const session = this[kPendingSession];
+  this[kPendingSession] = null;
+  if (session) this.emit('session', session);
```

Comment:
Another one... should these be wrapped in `process.nextTick`/`queueMicrotask`?

Reply:
I don't think it is needed. Will look into this once we have more test coverage.